File: pdu/python/hako_msgs/pdu_cdr_conv_DisturbanceUserCustom.py
from .pdu_pytype_DisturbanceUserCustom import DisturbanceUserCustom
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_DisturbanceUserCustom(writer: CdrWriter, src: DisturbanceUserCustom):
    writer.write_sequence_length(src.data)
    for elem in src.data:
        writer.write_float64(elem)


def cdr_body_to_py_DisturbanceUserCustom(reader: CdrReader, dst: DisturbanceUserCustom):
    dst.data = [reader.read_float64() for _ in range(reader.read_uint32())]
    return dst


def py_to_cdr_DisturbanceUserCustom(src: DisturbanceUserCustom) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_DisturbanceUserCustom(writer, src)
    return writer.bytes()


def cdr_to_py_DisturbanceUserCustom(cdr_payload) -> DisturbanceUserCustom:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = DisturbanceUserCustom()
    return cdr_body_to_py_DisturbanceUserCustom(reader, dst)

File: pdu/javascript/hako_msgs/pdu_cdr_conv_DisturbanceUserCustom.js
import { DisturbanceUserCustom } from './pdu_jstype_DisturbanceUserCustom.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduDisturbanceUserCustomConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { DisturbanceUserCustom } src
     */
    static to_cdr_body(writer, src) {
        writer.write_sequence_length(src.data);
        for (const elem of src.data) {
            writer.write_float64(elem);
        }
    }

    /**
     * @param {PduCdrReader} reader
     * @param { DisturbanceUserCustom } dst
     * @returns { DisturbanceUserCustom }
     */
    static cdr_body_to_js(reader, dst) {
        dst.data = Array.from({ length: reader.read_uint32() }, () => reader.read_float64());
        return dst;
    }

    /**
     * @param { DisturbanceUserCustom } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { DisturbanceUserCustom }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new DisturbanceUserCustom());
    }
}
